### PR TITLE
Add SPM Support

### PR DESCRIPTION
**Pull request by scannillo**
## Changes
* Add Package.swift
* Update header files to satisfy SPM (see the[ DEVELOPMENT.md](https://github.com/braintree/braintree_ios/blob/master/DEVELOPMENT.md#importing-header-files) for `braintree_ios` which explains this)
   * Update import syntax when needed, to use modular style imports
   * Nest public headers in `BraintreeVisaCheckout` directory (also explained in the DEVELOPMENT.md linked above)
* Add SPMTest app for testing on CI (and manually when needed)
* Add CI build for SPM, using SPMTest sample app
* Update CHANGELOG & README

### CODE REVIEW

**Comment by scannillo** [comment]
_UI test expected to fail until we get updated test account credentials._

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
   * Require `braintree_ios` major version v5
   * Replace `VisaCheckout.framework` v7.2.0 with `VisaCheckout.xcframework` v7.3.1
   * Drop Carthage support (Carthage support wasn't working in the latest 2 major versions)
+* Add Swift Package Manager support
```
Comment (sarahkoop) [inline CHANGELOG.md:8]: Is moving the header files considered a breaking change?

**Review by sarahkoop — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -4,29 +4,24 @@ Welcome to Braintree's iOS Visa Checkout SDK. This library will help you accept
 
 **The Braintree iOS Visa Checkout SDK is currently in a limited release and the API is subject to change.**
 
-**The Braintree iOS Visa Checkout SDK requires Xcode 12+**. It permits a Deployment Target of iOS 12.0 or higher.
+**The Braintree iOS Visa Checkout SDK requires Xcode 13+**. It permits a Deployment Target of iOS 12.0 or higher.
 
 ## Getting Started
 
-We recommend using [CocoaPods](https://github.com/CocoaPods/CocoaPods) to integrate the Braintree iOS Visa Checkout SDK with your project.
+We recommend using [Swift Package Manager](https://swift.org/package-manager/) or [CocoaPods](https://github.com/CocoaPods/CocoaPods) to integrate the BraintreeVisaCheckout SDK with your project.
```
Comment (sshropshire) [inline README.md:11]: 🔥

**Review by sshropshire — APPROVED** [review]
🌊

File: CHANGELOG.md (hunk 1 above)
Comment (scannillo) [reply CHANGELOG.md:8]: We're releasing a major version, so we can do that! And honestly - I don't think it would constitute a breaking change anyways

**Pull request merged**